### PR TITLE
Added shuffle option to all flow methods for GraphSAGE and HinSAGE generators

### PR DESCRIPTION
**Pull request by adocherty**
Previously, as we use `fit_generator` the mini-batches created by our generators have examples that are in fixed order. This will likely reduce the performance of the trained model, especially when our batch sizes are small.

This PR enables the generators to shuffle the data at each epoch when the `shuffle=True` option is given to the `.flow` methods of the generator object.

The notebooks have been updated to use the `shuffle=True` option when training and not use it at prediction time.

### CODE REVIEW

**Comment by adocherty** [comment]
Additionally, I've updated the `requirements.txt` file to fix the current documentation build failure on readthedocs, and removed `pyproject.toml` due to PEP-517 errors when `pip` reads the file – this file was used by poetry, but is now breaks newer versions of `pip` which read this file.

File: tests/mapper/test_link_mappers.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def example_Graph_2(feature_size=None):
     # Add example features
     if feature_size is not None:
         for v in G.nodes():
-            G.node[v]["feature"] = np.ones(feature_size)
+            G.node[v]["feature"] = int(v) * np.ones(feature_size)
```
Comment (youph) [inline tests/mapper/test_link_mappers.py]: what does this do?

[the remark this answers is not in the document]
Comment (adocherty) [reply tests/mapper/test_link_mappers.py]: This assigns a feature with the value of the node_id (it assumes node IDs are integers, which are true for current tests). This is so I can check which node a feature has come from when I test shuffling.

**Comment by adocherty** [comment]
This post was useful in implementing the shuffling:
https://stanford.edu/~shervine/blog/keras-how-to-generate-data-on-the-fly

However, We don't really need to turn shuffling off in the `fit_generator` call – shuffling the batches will have no effect after all examples are shuffled, so it's not necessary to set `shuffle=False`.

I think for the full batch node generator we can just not have the option, as it isn't applicable.  I think having the option and then complaining about it if it's `True` is more confusing.
